### PR TITLE
REF: organize DataFrame tests for indexing, by-method

### PR DESCRIPTION
**Pull request by jbrockmendel**
The only change that isnt just rearrangement is implementing where_frame fixture in test_where and using it to get rid of for-loops inside a bunch of tests in that file

### CODE REVIEW

File: pandas/tests/frame/indexing/test_where.py [hunk 1]
```diff
@@ -10,22 +10,30 @@
...
-                    issubclass(s.dtype.type, (np.integer, np.floating))
-                    and s.dtype != "uint8"
-                )
-
-            return DataFrame(
-                dict((c, s + 1) if is_ok(s) else (c, s) for c, s in df.items())
-            )
+@pytest.fixture(params=["default", "float_string", "mixed_float", "mixed_int"])
+def where_frame(request, float_string_frame, mixed_float_frame, mixed_int_frame):
+    if request.param == "default":
```
Comment (jreback) [inline pandas/tests/frame/indexing/test_where.py]: in future can add a doc-string (as the fixtures show this when showing the fixtures in pytest)

File: pandas/tests/frame/indexing/test_xs.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+import pandas as pd
+from pandas import DataFrame, Series
+import pandas._testing as tm
+
+from pandas.tseries.offsets import BDay
+
+
+class TestXS:
+    def test_xs(self, float_frame, datetime_frame):
```
Comment (jreback) [inline pandas/tests/frame/indexing/test_xs.py]: no big deal, but happy to have no classes if this is a specific test file (like you have it now).

**Review by jreback — APPROVED** [review]
lgtm. some follown comments. ping on green.

**Comment by jbrockmendel** [comment]
ping

**Comment by jreback** [comment]
thanks!

**Pull request merged**